IPA-End-to-End.md

```diff
@@ -1924,6 +1927,46 @@ An example of an oblivious algorithm is the bitonic sorting algorithm [[1]](http
 The last-touch attribution algorithm described above is oblivious. The algorithm can be parallelized into a large number of small computations that only compare two rows at a time. The pairs of rows which will be compared are always the same, regardless of the values of the input. As such, the _helper party_ learns nothing about the input, and can easily parallelize all of these comparisons.
 
 
+## Webviews and Embedded Browsers
+
+On some operating systems, browser components can be integrated into applications in a number of ways.
+Two models are considered here:
+
+1. Where the embedded browser component permits the application to access (or modify) content.
+
+2. Where the operation of the browser component is not accessible to the application.
+
+### Applications that can access web content
+
+If the embedded browser makes content accessible to the application, then conceptually the application needs to act as an independent browser.
+The application should have distinct storage from the regular browser so that the application is unable to access cookies or other data that is shared by users and sites.
+
+Note that this is an imperfect model because users can be presented with login pages on sites, which they might use.
+Application often use embedded browsers to manage login and so people are accustomed to having to log in, even if that is not a great security practice.
+
+When IPA integrates into this sort of environment, the `getEncryptedMatchKey()` API ideally returns a value that is specific to the application, not the active site.
+This approach would require operating system integration and some means of identifying the application that is compatible with the registrable domain used to identify sites on the web.
+The main drawback being that sites that are not aware of this constraint will not be able to use the value they obtain as it is bound to the application.
+
+Without OS integration that at least provides the embedded browser with a way to identify the application, the IPA API might need to be disabled.
+
+This can be partly addressed by making the site aware of the fact that it is embedded this way.
+Sites can then participate in attribution as before, provided that events that the site is associated with the event by the _report collector_.
+For instance, by setting a different _breakdown key_, the _record collector_ could allow in-app impressions to be aggregated separately to in-browser impressions.
+
+A malicious application might attempt to claim credit for conversions based on impressions in the embedded application.
+We offer no mitigation for this attack as there is no possible defense.
+There is no way to distinguish between an application that embeds a web view and one that provides a complete browser implementation of its own.
```

Comment:
I think we should change to: "This is a complex topic and we haven't yet done a thorough analysis"

Reply:
The other issue here we could callout is that if the webview could bind the reports for an event as if they happened on multiple sites, then the app/webview could setup budget duplication attacks.